datafusion/physical-expr/src/aggregate/sum.rs

```diff
@@ -53,11 +56,12 @@ impl Sum {
         name: impl Into<String>,
         data_type: DataType,
     ) -> Self {
-        let data_type = sum_return_type(&data_type).unwrap();
+        let return_type = sum_return_type(&data_type).unwrap();
```

Comment:
`sun_return_type` doesn't cover the `List` type for now

https://github.com/apache/arrow-datafusion/blob/2ffda2a9a893455e55cd773d9dd4f426a61d8cd3/datafusion/expr/src/type_coercion/aggregates.rs#L380-L399